test/framework/toy_build.py

```diff
@@ -563,23 +565,64 @@ def test_toy_gid_sticky_bits(self):
             self.assertTrue(perms & stat.S_ISVTX, "sticky bit set on %s" % fullpath)
 
     def test_toy_group_check(self):
-        """Test the userInGroup function with lua modules"""
-        toy_ec_file = os.path.join(os.path.dirname(__file__), 'easyconfigs', 'test_ecs', 't', 'toy', 'toy-0.0-group.eb')
+        """Test presence of group check in generated (Lua) modules"""
+        fd, dummylogfn = tempfile.mkstemp(prefix='easybuild-dummy', suffix='.log')
+        os.close(fd)
+
+        # figure out a group that we're a member of to use in the test
+        out, ec = run_cmd("groups $USER", simple=False)
+        self.assertEqual(ec, 0, "Failed to select group to use in test")
+        group_name = out.split(' ')[0]
+
+        toy_ec = os.path.join(os.path.dirname(__file__), 'easyconfigs', 'test_ecs', 't', 'toy', 'toy-0.0.eb')
+        test_ec = os.path.join(self.test_prefix, 'test.eb')
         args = [
-            '--sourcepath=%s' % self.test_sourcepath,
-            '--buildpath=%s' % self.test_buildpath,
-            '--installpath=%s' % self.test_installpath,
-            '--debug',
-            '--unittest-file=%s' % self.logfile,
+            test_ec,
             '--force',
             '--module-only',
         ]
 
-        # test specifying a Tcl module syntax
-        allargs = [toy_ec_file] + args + ['--module-syntax=Tcl']
-        outtxt, err = self.eb_main(allargs, logfile=self.dummylogfn, do_build=True, return_error=True)
-        err_regex = re.compile("Can't generate robust check in TCL modules for users belonging to group randomgroup.")
-        self.assertTrue(err_regex.search(outtxt), "Pattern '%s' found in '%s'" % (err_regex.pattern, self.dummylogfn))
+        for group in [group_name, (group_name, "Hey, you're not in the '%s' group!" % group_name)]:
+
+            if isinstance(group, basestring):
+                write_file(test_ec, read_file(toy_ec) + "\ngroup = '%s'\n" % group)
+            else:
+                write_file(test_ec, read_file(toy_ec) + "\ngroup = %s\n" % str(group))
+            outtxt, _ = self.eb_main(args, logfile=dummylogfn, do_build=True, return_error=True)
+
+            if get_module_syntax() == 'Tcl':
```

Comment:
Sure. But if you have a recent Lmod you can test all cases. Instead, we need now to pushing the PR and wait for travis to report that in certain case that you can test (but you didn't) it fails. Seems unnecessary to me.